src/serviceDesks/incontact/inContactServiceDesk.ts

```diff
@@ -146,9 +132,24 @@ class InContactServiceDesk implements ServiceDesk {
         const output = await request.json();
         if (output.status) {
           switch (output.status) {
-            case 'Waiting':
+            case 'Active':
               this.callback.agentJoined(this.agent);
               break;
+            case 'Waiting':
+              try {
+                const request = await fetch(`${this.SERVER_BASE_URL}/incontact/queue`, {
+                  method: 'POST',
+                  headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
+                  body: JSON.stringify({ ...this.session }),
+                });
+                const output = await request.json();
+                if (output && output.queue != undefined) {
```

Comment:
Also optional chaining. And do you need a `undefined` check or should this just be a falsy check?

Also
![image](https://user-images.githubusercontent.com/25828402/118148494-3c3dfa00-b3d6-11eb-85f4-2a23b81890bd.png)

Reply:
We need an undefined check here because the queue value can be 0 (if you're first in line, and the agent hasnt responded yet, the queue is 0), which evaluates to false. I added a comment for clarity here, and i also incremented the number that we show on our widget to add 1, because I think it makes more logical sense to show that you're 1st in line when the queue is empty.